tests/fs/ioctl.rs

```diff
@@ -19,6 +19,7 @@ fn test_ioctl_ficlone() {
     // that don't support this.
     match rustix::fs::ioctl_ficlone(&dest, &src) {
         Ok(()) | Err(io::Errno::OPNOTSUPP) => (),
+        Err(e) if e == io::Errno::from_raw_os_error(0x12) => (),
```

Comment:
Unrelated to this PR; I keep getting `EXDEV` for this test and I don't know why.

Reply:
That looks like a bug in the test. `src` is "Cargo.toml" and `dest` is a `tempfile()` so if you have a tmpfs on /tmp or so, that does sound like it could get an `EXDEV`. We should change that code to create two files in `dir` and then call `ioctl_ficlone` on those files.